### ISSUE TITLE
Design System Staging Review - Accessibility Feedback - Minimal Header

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#70359]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- **Thing to double-check:** Verify that the `banner` role is correctly announced in VoiceOver on Mac. I don't have a Mac to test on, but it seems fine in NVDA and JAWS. I know some screen readers are inconsistent in how they pass along ARIA attributes on web components, but I don't think we've had any web components with a `role` prior to this.
- **Vague question that may be outside the scope of the DST:** Not 100% sure how the `title` attribute gets generated on each page of VA.gov, but with the form name moving into the minimal header (which is currently the `title` for most forms), just want to make sure the `title` gets built with something sensible. My thought would be *[H1 text] | [Form name] | Veterans Affairs* or something like that.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Must:** Right now, the component is coded with up to two headings inside the minimal header. Example:
   ```
   <div class="header-container">
      <h1>Authorization To Disclose Personal Information To A Third Party</h1>
      <h2>(VA Form 21-0845)</h2>
   </div>
   ```
   Part of the goal for the forms team is to make the H1 in the page content be the step you're currently on, eg. "Your name and date of birth," rather than each step of the form having the same H1 (the name of the form). While technically allowed in the HTML spec, multiple H1s on the page can be confusing for screen reader users. I think the best option here is to remove the `h1` and `h2` from this component and use a `span` or similar. The heading semantics aren't really necessary here, and the `banner` role should make the purpose of the component clear to screen reader users.

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/va.gov-team/70359?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc

**Comment by rsmithadhoc** [comment]
- I checked `role="banner"` on VoiceOver, that shows up correctly as a landmark.
![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/100248909/72af385a-b5f2-415e-b906-f8a0630b52b5)
- For the page title, that will fall to the team implementing the page with this component. 
- For the **Must** feedback, I opened a PR https://github.com/department-of-veterans-affairs/component-library/pull/1084. 
  - I believe we implemented it based on the design, without the use case for forms in mind. I agree that it would need the headings removed, so I added an optional prop to disable the headings.
